app/src/main/java/org/literacyapp/authentication/TrainingHelper.java

```diff
@@ -243,6 +256,17 @@ public synchronized void trainClassifier(){
                         studentImageCollectionEvent.setSvmTrainingExecuted(true);
                         studentImageCollectionEventDao.update(studentImageCollectionEvent);
                         Log.i(getClass().getName(), "StudentImageCollectionEvent with Id " + studentImageCollectionEvent.getId() + " has been trained in classifier");
+                    } else if (TextUtils.isEmpty(studentImageCollectionEvent.getStudent().getAvatar())){
+                        // Try to create an Avatar for the Student if no Avatar has been created yet
```

Comment:
@sladomic If the Student's `avatar` property is `@NotNull`, will this code ever be triggered?

https://github.com/literacyapp-org/literacyapp-android/blob/master/app/src/main/java/org/literacyapp/model/Student.java#L42

Ideally a Student without an image should never be stored in the database in the first place, right? (`studentDao.insert(student);` will fail).